### ISSUE TITLE
File Name Conventions for SA Mobile 1.06+ and SA Remastered PS3

### ISSUE THREAD

**Issue by OrionSR** [sannybuilder/dev#69]
The file name convention for main.scm and script.img is different for SA mobile since apk106 and for the PS3.

Mobile uses mainV1.scm and scriptv1.img
PS3 uses mainPS3.scm and scriptps3.img

Formatting settings do not appear to be flexible enough to manage the native filenames, and any changes would have a global impact.
```[Formats]
N1=$dir\$name.txt
N2=$dir\$name.scm
N3=$dir\script.img
```
The user would need to manage the filename of script image to work with the files. I usually just make a copy of scriptv1.img as script.img the first time I decompile the main script and forget about it. I don't have a way to implement a modified scriptv1.img but others do. 

Also, Sanny throws an error if it can't find the script.img associated with main.scm when decompiling but there doesn't appear to be a name setting for decompiling the img file.

This issue hasn't been a major irritation. The copied file works well enough that I forget about the problem. But since I stumbled across it again I figured I'd better mention it while I'm thinking about it.

**Comment by OrionSR** [comment]
This issue has been handled gracefully in SB 3.80. When the image name was ambiguous I was prompted to choose a file from a list.

**Comment by OrionSR** [comment]
I took another look; the primary issue remains, but is better now because when script.img is missing I'm prompted to choose a file rather than required to copy or rename the file before trying to decompile again.

The test case for DE was unusual. Usually I keep my mains in different folders so they can't get confused or mixed. The leaked scripts for DE had 3 versions in the same data\script\ folder; main.scm(unk), mainV1.scm(DE) and mainPS3.scm, with img files for V1 and PS3. Script.img was missing, probably due to a bug in the V1 batch file.

The unknown main can't be decompile without matching it up with a known image, but instead I just threw in an image from compiling PC repros with CS. When I tried to decompile mainPS3 SB tried to use the bogus script.img and encountered an unknown opcode error. The odd part here is, I was expecting SB to throw the usual image is broken or wrong message. Actually decompiling it with IGNORE_UNKNOWN was unexpected.

It's easy to predict how compiling different versions of main in the same folder will likely lead to mismatched images, because we understand how it works. But the association between .scm and .img files is so obvious that a user would reasonably expect SB to figure it out, globally. Something like:

'N3=$dir\$substitute($name,"main","script").img'

**Comment by x87** [comment]
I guess I can just add a directive that sets the img archive name. 

```
{$IMG scriptps3.img}
```

it will override the default rules from Option->Formats

**Comment by OrionSR** [comment]
Nah... That wouldn't solve the problems of choosing a matching img or incorrectly loading script.img if present. Better to wait for a nice global solution, especially if I'm the only one worried about it. I can manage my files effectively enough. Also, recompiling mains for mobile, PS3 and DE doesn't really work well, or at all, so this is a fairly specialized task. Mobile mains need to be repacked in the apk to be implemented, no idea how this could work on PS3, and reports suggest lots of trouble implementing custom mains on DE, especially the custom scriptv1.img files. 

How about a new formatting option; Automatic filename management? Leave the old implementation as it's been but add a toggle to let SB manage the file names as expected for decompiling and compiling.

*main*.txt = *main*.scm + *script*.img
main.txt = main.scm + script.img
mainV1.txt = mainV1.scm + scriptv1.img
mainPS3.txt = mainPS3.scm + scriptps3.img
MyMain.txt = MyMain.scm + myscript.img
TTDISA.txt = TTDISA.scm + ttdisa.img

Give it some time and players will eventually figure out how to get custom mains working on DE. Sounds like there's no need to rush.

**Comment by x87** [comment]
right, but if you decompile MyMain.scm you would get `{$IMG  myscript.img}` and then the compiler would be picking this name and making MyMain.scm (because of the $name.scm rule) and `myscript.img`. And you are still given an option to customize this directive's value for your needs.
